zk-token-sdk/src/encryption/discrete_log.rs

```diff
@@ -1,18 +1,16 @@
 #![cfg(not(target_arch = "bpf"))]
 
 use {
-    curve25519_dalek::{ristretto::RistrettoPoint, scalar::Scalar, traits::Identity},
+    curve25519_dalek::{
+        constants::RISTRETTO_BASEPOINT_POINT as G, ristretto::RistrettoPoint, scalar::Scalar,
+        traits::Identity,
+    },
     serde::{Deserialize, Serialize},
-    std::collections::HashMap,
+    std::{collections::HashMap, sync::mpsc, thread},
 };
 
-#[allow(dead_code)]
-const TWO15: u64 = 32768;
-#[allow(dead_code)]
-const TWO14: u64 = 16384; // 2^14
 const TWO16: u64 = 65536; // 2^16
-#[allow(dead_code)]
-const TWO18: u64 = 262144; // 2^18
+const NUM_THREADS: usize = 1;
```

Comment:
🤔 How does the user override this without modifying the source to this crate?

Reply:
Yes, this has to be adjusted later during integration. For this PR, I thought I would just enable multithreading capability using a constant variable. Or is there a conventional way for users to set the variable?

Reply:
Oh, I guess the best way is for the decryption function to take in the number of threads as a parameter with default set to 1. Maybe that is what you were pointing out... sorry.